### PR TITLE
etcdctl: use cluster endpoints when passed --cluster

### LINKED ISSUE #8117: There're no cluster level command to show all the members' status/health in V3
## Problem state

In V2, the command `etcdctl cluster health` checks all the members' health status. And in V3, there's 1 command, 2 sub-commands to check the member's health status - `etcdctl endpoint status` and `etcdctl endpoint health`. 
However, the V3 commands are difficult to use cause they only check the endpoints' status/health provided by option `--endpoints`. It is needed to collect all the endpoints first, and then set into the endpoints option.

There's one [issue created #8115 ](https://github.com/coreos/etcd/issues/8115) first to implement the feature to collect all the members' status/health.

It is supposed to be the design instead of implementation. I hope there is one command to check the cluster status instead of only the endpoints provided.

```
$ ETCDCTL_API=3 etcdctl cluster health -w table
+----------------------------------------------+------------------+---------+---------+-----------+-----------+------------+
|                   ENDPOINT                   |        ID        | VERSION | DB SIZE | IS LEADER | RAFT TERM | RAFT INDEX |
+----------------------------------------------+------------------+---------+---------+-----------+-----------+------------+
| https://test-node-vh8tr-6888.51.test.cn:4001 | d479908fcc05ba8e | 3.0.15  | 8.0 GB  | false     |      1309 |  229422110 |
| https://test-node-2o9hu-4886.51.test.cn:4001 | b620b4c395187fad | 3.1.8   | 8.0 GB  | false     |      1309 |  229422110 |
| https://test-node-873t0-8911.51.test.cn:4001 | 829de42e4f2097c4 | 3.1.8   | 8.0 GB  | true      |      1309 |  229422110 |
| https://test-node-ba93c-3871.51.test.cn:4001 | 2af130f12df09e15 | 3.0.15  | 8.0 GB  | false     |      1294 |  227711328 |
| https://test-node-xq2sm-5915.51.test.cn:4001 | 5902a07919e43cdf | 3.0.15  | 8.0 GB  | false     |      1309 |  229422110 |
+----------------------------------------------+------------------+---------+---------+-----------+-----------+------------+
```

The confusion part is that there is already one command which can make this.
```
$ ETCDCTL_API=3 etcdctl endpoint status --endpoints=https://test-node-vh8tr-6888.51.test.cn:4001,https://test-node-2o9hu-4886.51.test.cn:4001,https://test-node-873t0-8911.51.test.cn:4001,https://test-node-ba93c-3871.51.test.cn:4001,https://test-node-xq2sm-5915.51.test.cn:4001 -w table
....
```

I propose to change the logic of `endpoint status`, `endpoint health` to collect all the endpoints and check the health of them. It is implemented by PR: https://github.com/coreos/etcd/pull/8116

### PR DESCRIPTION
**Pull request by heyitsanthony**
Query the cluster for endpoints when given --cluster for the endpoint commands.

Fixes #8117

### CODE REVIEW

File: etcdctl/README.md [hunk 1]
```diff
@@ -565,6 +565,10 @@ Prints a humanized table of the member IDs, statuses, names, peer addresses, and
 
 ENDPOINT provides commands for querying individual endpoints.
 
+#### Options
+
+- cluster -- fetch and use all endpoints from the etcd cluster
```
Comment (xiang90) [inline etcdctl/README.md]: from the etcd cluster membership?

File: etcdctl/ctlv3/command/ep_command.go [hunk 2]
```diff
@@ -139,3 +138,29 @@ func epStatusCommandFunc(cmd *cobra.Command, args []string) {
...
+	}
+	c := mustClientFromCmd(cmd)
+	ctx, cancel := commandCtx(cmd)
+	defer func() {
+		c.Close()
+		cancel()
+	}()
+	membs, err := c.MemberList(ctx)
+	if err != nil {
+		ExitWithError(ExitError, err)
```
Comment (xiang90) [inline etcdctl/ctlv3/command/ep_command.go]: have you checked what the error looks like?

can we provide more context here: "failed to get all endpoints from etcd cluster membership: %v"

File: etcdctl/README.md (hunk 1 above)
Comment (xiang90) [inline etcdctl/README.md]: this also suffers from the same issue as the previous `auto-sync` option. if users have a proxy (not necessarily etcd proxy or etcd gPRC proxy) in front of etcd, and they can only access the cluster though the proxy, then cluster option will try to bypass the proxy. this is not the desired behavior. we probably need to warn about that here?

**Comment by xiang90** [comment]
lgtm except some document improvement suggestions.

File: etcdctl/README.md (hunk 1 above)
Comment (heyitsanthony) [inline etcdctl/README.md]: It says it's fetching it from the cluster though; if there's a pass-through proxy then the cluster wouldn't know about it. Would "fetch and use all endpoints from the etcd cluster member list" be OK?

Comment (xiang90) [inline etcdctl/README.md]: ok. let us make it this way. if users have issue, we can improve.

**Pull request merged**